crates/matrix-sdk-crypto/src/olm/group_sessions/sender_data.rs

```diff
@@ -89,14 +88,68 @@ impl SenderData {
         }
     }
 
+    /// Create a [`SenderData`] which contains device info.
+    pub fn device_info(device_keys: DeviceKeys) -> Self {
+        Self::DeviceInfo {
+            device_keys,
+            // TODO: when we have implemented all of SenderDataFinder,
+            // legacy_session should be set to false, but for now we leave
+            // it as true because we might lose device info while
+            // this code is still in transition.
+            legacy_session: true,
+        }
+    }
+
+    /// Create a [`SenderData`] with a known sender.
+    pub fn sender_known(
+        user_id: &UserId,
+        device_id: &DeviceId,
+        master_key: Ed25519PublicKey,
+        master_key_verified: bool,
+    ) -> Self {
+        Self::SenderKnown {
+            user_id: user_id.to_owned(),
+            device_id: Some(device_id.to_owned()),
+            master_key: Box::new(master_key),
+            master_key_verified,
+        }
+    }
+
     /// Create a [`SenderData`] which has the legacy flag set. Caution: messages
     /// within sessions with this flag will be displayed in some contexts,
     /// even when we are unable to verify the sender.
     ///
-    /// The returned struct contains no device info, and will be retried soon.
+    /// The returned struct contains no device info.
     pub fn legacy() -> Self {
         Self::UnknownDevice { legacy_session: true, owner_check_failed: false }
     }
+
+    /// Returns true if this is SenderKnown and `master_key_verified` is true.
+    pub(crate) fn is_known_and_verified(&self) -> bool {
+        matches!(self, SenderData::SenderKnown { master_key_verified: true, .. })
+    }
+
+    /// Returns true if this SenderData represents a greater level of trust than
+    /// the supplied one.
+    ///
+    /// So calling this method on a SenderKnown or DeviceInfo SenderData would
+    /// return true if passed an UnknownDevice as its argument, and a
+    /// SenderKnown with `master_key_verified == true` would return true if
+    /// passed a SenderKnown with `master_key_verified == false`.
+    ///
+    /// Returns false if the two SenderDatas have the same trust level.
+    pub(crate) fn is_more_trusted_than(&self, other: &Self) -> bool {
```

Comment:
Probably overly pendantic, but do you mind of introducing a type for the result of the method instead of using a boolean, this would require the function to be renamed, but that way we don't fold the `==` case into false. We have something similar for the comparison of the quality of inbound group sessions:

https://github.com/matrix-org/matrix-rust-sdk/blob/21efd60dbab1c16f9cec3a77d88429601ef645d7/crates/matrix-sdk-crypto/src/store/mod.rs#L1023-L1038

Reply:
Changed to return an `Ordering` in be16e70391a4e2d3665bcb095147637c2b6f19c8 - I think that makes sense here... I created a custom enum and its names were almost identical to `Ordering`...